org-code-javabuilder/neighborhood/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -143,12 +143,20 @@ public void takePaint() {
 
   /** @return True if there is paint in the square where the painter is standing. */
   public boolean isOnPaint() {
-    return this.grid.getSquare(this.xLocation, this.yLocation).hasColor();
+    boolean isOnPaint = this.grid.getSquare(this.xLocation, this.yLocation).hasColor();
+    this.outputAdapter.sendMessage(
+        new NeighborhoodSignalMessage(
+            NeighborhoodSignalKey.IS_ON_PAINT, this.getSignalDetailsForBooleanMethod(isOnPaint)));
```

Comment:
not sure how strongly i feel about this, but all of the calls to `sendMessage` look pretty similar, so we could add a helper to simplify them like

```java
public void sendBooleanMessage(NeighborhoodSignalKey key, boolean value) {
  this.outputAdapter.sendMessage(
    new NeighborhoodSignalMessage(key, this.getSignalDetailsForBooleanMethod(value));
  );
}

public boolean isOnPaint() {
  this.sendBooleanMessage(NeighborhoodSignalKey.IS_ON_PAINT, isOnPaint);
}
```

Reply:
updated!